Data/ByteString/Short/Internal.hs

```diff
@@ -482,6 +487,48 @@ toShortIO (BS fptr len) = do
     touchForeignPtr fptr
     ShortByteString <$> stToIO (unsafeFreezeByteArray mba)
 
+-- | A simple wrapper around 'fromShort' that wraps the strict 'ByteString' as
+-- a one-chunk lazy 'LBS.ByteString'.
+lazyFromShort :: ShortByteString -> LBS.ByteString
+lazyFromShort = LBS.fromStrict . fromShort
+
+-- | /O(n)/. Convert a lazy 'LBS.ByteString' into a 'ShortByteString'.
+--
+-- This makes a copy, so does not retain the input string.  Naturally, best
+-- used only with sufficiently short lazy ByteStrings.
+--
+lazyToShort :: LBS.ByteString -> ShortByteString
+lazyToShort LBS.Empty = empty
+lazyToShort (LBS.Chunk c@(BS _ len) cs) =
+    unsafeDupablePerformIO $ do
+        mba0 <- stToIO (newByteArray len)
+        mba <- lazyToShortIO mba0 0 len c cs
+        ShortByteString <$> stToIO (unsafeFreezeByteArray mba)
+
+lazyToShortIO :: MutableByteArray RealWorld
+              -> Int
+              -> Int
+              -> ByteString
+              -> LBS.ByteString
+              -> IO (MutableByteArray RealWorld)
+lazyToShortIO mba used size c@(BS fptr len) cs
+    | len <= size - used
+      = do let ptr = unsafeForeignPtrToPtr fptr
+           stToIO (copyAddrToByteArray ptr mba used len)
+           touchForeignPtr fptr
+           case cs of
+               LBS.Empty
+                   -> stToIO (shrinkByteArray mba $ used + len) >> pure mba
+               LBS.Chunk c' cs'
+                   -> lazyToShortIO mba (used + len) size c' cs'
+    | used < used + len
+    , let newsize = max (used + len) (size + size `div` 2)
```

Comment:
Generally speaking, it's better to use `quot` instead of `div` or just an explicit `shift`. In this case maybe `size * 2` would work as well?..

Reply:
The reason for the factor of `1.5` rather than 2 is the somewhat well known observation that when resizing by factors less than the golden ratio, after a few copies the total memory released exceeds the new memory requested, and one may be able to reuse the space.

Unclear how applicable this is to GHC's memory allocator, but this is at least harmless, and still ensures that the allocation cost repeated resizing for large objects is not quadratic.  So I'll stick with 1.5, but sure, I'll change it to `quot`.